### DISCUSSION TITLE
Functional Difference Between appimage and flatpak versions? (Specifically related to nnn fifo)

### DISCUSSION THREAD

**Discussion by levone1** [wezterm/wezterm#4890] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

kwin_X11

### WezTerm version

20240127-113634-bbcac864

### Ask your question!

Just curious, (I could easily work-around)...

Using nnn file manager, and generally prefer to use it with preview-tui plugin running.  Plugin depends on fifo, which can be set in env variable, (export NNN_FIFO), or with '-a' subcommand, (auto-fifo).  Have generally used it with Kitty with no trouble, and recently started getting faimliarized with Wezterm via appimage file, (placing appimage file in a "bin" directory, and creating a desktop file), and it also worked fine, (preview plugin worked with no error).

Then, I decided to try the flatpak version, (the first flatpak I have installed).  It installed and opens fine, and when I launch nnn, it launches fine, but no matter what I've tried, the preview plugin won't work; I just get "no FIFO available" message, (which is the typical error message you get in nnn if you try to start the plugin without one of the previously mentioned conditions in place).

I've launched with "nnn -a", I've exported NNN_FIFO, (from command line and in ...rc file), and I've created the fifo manually but always the same.  "echo $NNN_FIFO" command shows the fifo set in env.

I know it's not the responsibility of Wezterm to see to it that other people's programs work the way I want them to on my computer, but, like I said - just curious...  I thought maybe there would be some insight here about it.

I also have built Wezterm Cargo package, and it also worked fine with the preview plugin.  One thing I noticed about that is that the Cargo build process created 3 executables, and I think I usually launched the one called "wezterm-gui", so not sure if that means anything...  

So is there some functional difference in the different packages, or is it an issue specific to flatpak, or...?

Thanks

**Comment by wez** [comment, marked as answer]
flatpak runs programs in an isolated container. In order to spawn programs, wezterm needs to ask the flatpak runtime to spawn them on its behalf via a utility called `flatpak-spawn`.

I don't know the precise details or explanation for the interaction with the program you are trying to run, but I suspect that the flatpak-spawn'd process is not fully associated with the host namespaces and is thus not able to access the fifo.

Both the flatpak and AppImage versions of wezterm are provided primarily for ease of trying out the software, and I recommend that folks use a native package once they've decided to really adopt wezterm, so that they have full access to their host system.
